packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -326,11 +332,13 @@ where
                 },
                 account_update,
                 consensus_update: update.clone(),
-            });
+            };
+            headers.push(header.clone());
```

Comment:
nit: You can add the debug statement before this push and remove the clone if you want since pushing cannot fail anyway...